scene/gui/tab_container.cpp

```diff
@@ -479,7 +481,14 @@ void TabContainer::set_current_tab(int p_current) {
 	}
 
 	_change_notify("current_tab");
-	emit_signal("tab_changed",current);
+
+	if (pending_previous == current)
+		emit_signal("tab_selected", current);
```

Comment:
Well, the whole issue in #7752 is about that, `tab_selected` is emitted when selecting the same tab, and `tab_changed` is when selecting `!current` tab.